lib/reporter.js

```diff
@@ -0,0 +1,90 @@
+var Stream = require('stream');
+var Hoek = require('hoek');
+
+var internals = {};
+
+internals.GoodTransform = module.exports = function (events, options) {
+
+    options = options || {};
+    options.objectMode = true;
+
+    if (!(this instanceof internals.GoodTransform)) {
+        return new internals.GoodTransform(events, options);
+    }
+
+    Stream.Transform.call(this, options);
+    this._good = {
+        subscription: internals.GoodTransform.subscription(events)
+    };
+};
+
+
+Hoek.inherits(internals.GoodTransform, Stream.Transform);
+
+
+internals.GoodTransform.prototype._transform = function (data, enc, next) {
+
+    if (internals.GoodTransform.filter(this._good.subscription, data.event, data.tags)) {
+        this.push(data);
+    }
+    next(null);
+};
+
+
+// events hash of events and tags
+internals.GoodTransform.subscription = function (events) {
+
+    var result = Object.create(null);
```

Comment:
Since `events` comes from the user and we are iterating over the attached keys, I wanted to prevent issues if someone names a key in `events` something like "hasOwnProperty" and potentially causing downstream impacts.